File: clinic-admin/src/api/prescription.ts
import { get, post } from '@/utils/request'

export interface Prescription {
  id?: number
  prescriptionId?: number
  prescriptionNo?: string
  prescriptionType: number
  prescriptionTypeName?: string
  registrationId?: number
  patientId?: number
  patientName: string
  doctorId?: number
  doctorName?: string
  deptId?: number
  departmentName?: string
  diagnosis: string
  totalAmount?: number
  discountAmount?: number
  actualAmount?: number
  paymentStatus?: number
  statusName?: string
  items: PrescriptionItem[]
  createTime?: string
}

export interface PrescriptionItem {
  itemId?: number
  itemType: number
  itemTypeName?: string
  itemName: string
  spec?: string
  unit: string
  quantity: number
  unitPrice: number
  subtotal: number
  usage?: string
  frequency?: string
  duration?: string
  remark?: string
}

export interface PrescriptionQuery {
  clinicId: number
  doctorId?: number
  prescriptionType?: number
  paymentStatus?: number
  startDate?: string
  endDate?: string
  pageNum?: number
  pageSize?: number
}

export const getPrescriptionList = (params: PrescriptionQuery) =>
  get<{ total: number; list: Prescription[] }>('/admin/prescription/list', params)

export const getPrescriptionDetail = (prescriptionId: number) =>
  get<Prescription>(`/admin/prescription/${prescriptionId}`)

export const createPrescription = (data: Prescription) =>
  post<{ prescriptionId: number; prescriptionNo: string }>('/admin/prescription/create', data)

export const refundPrescription = (data: {
  prescriptionId: number
  operatorId: number
  refundReason: string
}) => post('/admin/prescription/refund', data)

export const dispensePrescription = (data: {
  prescriptionId: number
  operatorId: number
  remark?: string
}) => post('/admin/prescription/dispense', data)

File: clinic-api/src/main/java/com/clinic/entity/Prescription.java
package com.clinic.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.io.Serializable;
import java.math.BigDecimal;

@Data
@TableName("prescription")
public class Prescription implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long clinicId;
    private String prescriptionNo;
    private Integer prescriptionType;
    private Long registrationId;
    private Long patientId;
    private String patientName;
    private Long doctorId;
    private String doctorName;
    private Long deptId;
    private String departmentName;
    private String diagnosis;
    private BigDecimal totalAmount;
    private BigDecimal discountAmount;
    private BigDecimal actualAmount;
    private Integer paymentStatus;
    //private String chiefComplaint;
    //private java.time.LocalDateTime createTime;
    private java.time.LocalDateTime createdAt;
    private java.time.LocalDateTime updatedAt;
    private Integer deleted;

    /**
     * 缴费方式：1微信 2支付宝 3现金 4医保 5会员卡
     */
    private Integer paymentMethod;

    @com.baomidou.mybatisplus.annotation.TableField(exist = false)
    private String prescriptionTypeName;

    @com.baomidou.mybatisplus.annotation.TableField(exist = false)
    private String statusName;

    @com.baomidou.mybatisplus.annotation.TableField(exist = false)
    private java.util.List<PrescriptionItem> items;
}

